### PR TITLE
[WIP] refactor extract_2d 

### PR DESCRIPTION
**Pull request by nden**
This PR creates a `slit` schema and model. This new schema references the image schema and adds slit specific extensions - `wavelength` and all the pathloss extension. I am assuming the pathloss extensions were added to the `image` schema simply because there was no other specific schema to use (@stscirij?).  This PR removes them from the `image` schema but if they need to be I can add them back.

Also now extract_2d writes out a `wavelength` extension for use by other steps. The wavelength zero-point correction will be added in a separate PR. The main point of this refactoring is to improve performance. 
Every slit is created and validated separately  and then the list of slits is assigned to MultiSlitModel onnce.

todo:
[ ] documentation
[ ] testing

### CODE REVIEW

File: jwst/datamodels/__init__.py [hunk 1]
```diff
@@ -130,11 +131,12 @@
     'ReferenceFileModel', 'ReferenceCubeModel', 'ReferenceImageModel', 'ReferenceQuadModel',
     'RegionsModel', 'ResetModel',
     'ResolutionModel', 'MiriResolutionModel',
-    'RSCDModel', 'SaturationModel', 'SpecModel',
+    'RSCDModel', 'SaturationModel', 'SlitModel', 'SpecModel',
```
Comment (hbushouse) [inline jwst/datamodels/__init__.py]: 👍  for maintaining the alphabetical order of model definitions! This is the really important stuff.

File: jwst/datamodels/schemas/multislit.schema.yaml [hunk 2]
```diff
@@ -1,135 +1,10 @@
 allOf:
 - $ref: core.schema.yaml
-- $ref: bunit.schema.yaml
```
Comment (hbushouse) [inline jwst/datamodels/schemas/multislit.schema.yaml]: Is the definition for BUNIT keywords included again in some other place? We need those in the SCI and ERR extension headers.

File: jwst/extract_2d/extract_2d.py [hunk 3]
```diff
@@ -57,33 +59,40 @@ def extract2d(input_model, which_subarray=None):
...
+                new_model.source_alias = slit.source_alias
+                new_model.catalog_id  = slit.catalog_id
+                new_model.stellarity = float(slit.stellarity)
+                new_model.source_xpos = float(slit.source_xpos)
+                new_model.source_ypos = float(slit.source_ypos)
+                new_model.slitlet_id = int(slit.name)
                 # for pathloss correction
-                output_model.slits[nslit].nshutters = int(slit.nshutters)
+                new_model.nshutters = int(slit.nshutters)
+            slits.append(new_model)
```
Comment (hbushouse) [inline jwst/extract_2d/extract_2d.py]: Yay for only 1 append operation! 👍

**Comment by nden** [comment]
Superseded by #1070

**Pull request closed without merging**